net/client_grpc.go

```diff
@@ -54,10 +59,18 @@ func NewGrpcClientFromCertManager(c *CertManager, opts ...grpc.DialOption) Clien
 // method calls.
 func NewGrpcClientWithTimeout(timeout time.Duration, opts ...grpc.DialOption) Client {
 	c := NewGrpcClient(opts...).(*grpcClient)
+	c.loadEnvironment()
 	c.timeout = timeout
 	return c
 }
 
+func (g *grpcClient) loadEnvironment() {
+	opt := grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+		return proxy.Dial(ctx, "tcp", addr)
+	})
+	g.opts = append(g.opts, opt)
```

Comment:
Should this be added to the front of the options list so that a user can override it if they want?